### PR TITLE
Add SQLAlchemy integration

### PR DESCRIPTION
**Pull request by liyanhui1228**

### CODE REVIEW

File: trace/opencensus/trace/config_integration.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 log = logging.getLogger(__name__)
 
-SUPPORTED_INTEGRATIONS = ['mysql', 'postgresql']
+SUPPORTED_INTEGRATIONS = ['mysql', 'postgresql', 'sqlalchemy']
```
Comment (duggelz) [inline trace/opencensus/trace/config_integration.py]: What happens if I enable sqlalchemy and mysql or postgres integration at the same time?  Does everything get traced twice?  Should we add documentation and a warning for this case?

File: trace/opencensus/trace/ext/sqlalchemy/trace.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+from opencensus.trace import execution_context
+
+log = logging.getLogger(__name__)
+
+MODULE_NAME = 'sqlalchemy'
+
+
+def trace_integration():
+    """Integrate with SQLAlchemy to trace it using event listner.
```
Comment (duggelz) [inline trace/opencensus/trace/ext/sqlalchemy/trace.py]: listner -> listener

File: trace/opencensus/trace/config_integration.py (hunk 1 above)
Comment (liyanhui1228) [reply trace/opencensus/trace/config_integration.py]: The communication between SQLAlchemy and the database it connected will also be traced, and yes the query will be traced twice. Documentation added for this case.

File: trace/opencensus/trace/ext/sqlalchemy/trace.py (hunk 2 above)
Comment (liyanhui1228) [reply trace/opencensus/trace/ext/sqlalchemy/trace.py]: Fixed.

File: README.rst [hunk 3]
```diff
@@ -260,6 +260,19 @@ Run this command to install this package,
...
     pip install psycopg2
 
+SQLAlchemy
+~~~~~~~~~~
+
+Note that if enabled tracing both SQLAlchemy and the database it connected,
+the communication between SQLAlchemy and the database will also be traced.
+To avoid the verbose spans, you can just trace SQLAlchemy.
+
+Run this comman to install the SQLAlchemy package,
```
Comment (duggelz) [inline README.rst]: comman -> command

File: trace/tests/unit/ext/sqlalchemy/test_sqlalchemy_trace.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            'sqlalchemy/query/parameters': parameters,
+            'sqlalchemy/cursor/method/name': 'execute'
+        }
+
+        expected_name = '[sqlalchemy.query]{}'.format(query)
+
+        self.assertEqual(mock_tracer.current_span.labels, expected_labels)
+        self.assertEqual(mock_tracer.current_span.name, expected_name)
+
+    def test__before_cursor_executmany(self):
```
Comment (duggelz) [inline trace/tests/unit/ext/sqlalchemy/test_sqlalchemy_trace.py]: executmany -> executemany

File: README.rst (hunk 3 above)
Comment (liyanhui1228) [reply README.rst]: Fixed.

File: trace/tests/unit/ext/sqlalchemy/test_sqlalchemy_trace.py (hunk 4 above)
Comment (liyanhui1228) [reply trace/tests/unit/ext/sqlalchemy/test_sqlalchemy_trace.py]: Fixed.

**Pull request merged**